quodlibet/util/collection.py

```diff
@@ -598,25 +614,29 @@ def new(cls, dir_, base=_("New Playlist"), library=None):
             name = "%s %d" % (base, i) if i else base
             fn = cls.filename_for(name)
             try:
-                return cls(dir_, fn, library, validate=True)
+                return cls(dir_, fn, songs_lib=songs_lib, pl_lib=pl_lib, validate=True)
             except ValueError as e:
                 last_error = e
         raise ValueError("Couldn't create playlist of name '%s' (e.g. %s)"
                          % (base, last_error))
 
     @classmethod
-    def from_songs(cls, dir_, songs, library=None):
+    def from_songs(cls, dir_, songs, songs_lib=None, pl_lib=None):
         assert isinstance(dir_, fsnative)
         title = cls.suggested_name_for(songs)
-        playlist = cls.new(dir_, title, library)
+        playlist = cls.new(dir_, title, songs_lib=songs_lib, pl_lib=pl_lib)
         playlist.extend(songs)
         return playlist
 
     @property
     def path(self):
         return os.path.join(self.dir, self.filename_for(self.name))
 
-    def _validated_name(self, new_name):
+    @property
+    def key(self) -> str:  # type: ignore  # (Note: we want no setter)
```

Comment:
Isn't the "We want no setter" comment superfluous, since we see that there's no `@property.setter`?

Reply:
Yes, mypy gets annoyed if we don't add a setter, but we don't want one. I guess I could move the comment above perhaps to clarify